container-search/src/main/java/com/yahoo/search/Query.java

```diff
@@ -198,7 +198,7 @@ public static Type getType(String typeString) {
         argumentType.addField(new FieldDescription(HITS.toString(), "integer", "hits count"));
         // TODO: Should this be added to com.yahoo.search.query.properties.QueryProperties? If not, why not?
         argumentType.addField(new FieldDescription(SEARCH_CHAIN.toString(), "string"));
-        argumentType.addField(new FieldDescription(TRACE_LEVEL.toString(), "integer", "tracelevel"));
+        argumentType.addField(new FieldDescription(TRACE_LEVEL.toString(), "integer", "trace.level tracelevel traceLevel"));
```

Comment:
I don't think this works? If the name has a dot it really means there's a nested type under "trace", and you need to add both levels (same as with model, ranking etc.).

Reply:
Ah okay, but trace do not have it's own class like model, ranking etc. Should I make that, or 
create a QueryProfileType in Query, and with that adding new FieldDescriptions to the current QueryProfileType, like model, ranking etc?

I also see that trace.timestamps is a CompoundName, without a QueryProfileType in SearchHandler (ln 99). Should that also be added under "trace", as well as trace.rules in SemanticSearcher?

Reply:
I think it's good enough to add the QueryProfileType and no separate class.
Yes, trace.timestamps should be under that as well then.

Reply:
Tried to do this in latest commit.